### ISSUE TITLE
feat(inference): only-last-logit optimization for autoregressive generation

### ISSUE THREAD

**Issue by mrveiss** [mrveiss/AutoBot-AI#1968]
## Summary

During autoregressive generation, only compute the vocabulary projection (lm_head) for the last token position instead of all positions — a simple optimization that yields up to 2048x reduction in the final projection.

## Background

In standard generation, the lm_head linear layer projects hidden states to vocabulary logits. For a model with vocab_size=32000 and sequence_length=2048:

- **Without optimization**: `(batch, 2048, hidden) → (batch, 2048, 32000)` = ~125MB fp16 output
- **With optimization**: `(batch, 1, hidden) → (batch, 1, 32000)` = ~62KB fp16 output

Only the last position's logits determine the next token, so computing logits for all other positions is pure waste.

```python
def forward(self, hidden_states, only_last_logit=True):
    if only_last_logit:
        # 2048x reduction in lm_head computation + memory
        logits = self.lm_head(hidden_states[:, -1:, :])
    else:
        # Full projection (needed for perplexity scoring, etc.)
        logits = self.lm_head(hidden_states)
    return logits
```

## Key Files

- `autobot-backend/llm_interface_pkg/optimization/inference_utils.py` — new utility
- `autobot-backend/llm_interface_pkg/providers/` — apply in local inference providers

## Acceptance Criteria

- [ ] Only-last-logit mode enabled by default for generation
- [ ] Full logit mode available for perplexity scoring / evaluation
- [ ] Memory savings verified: ~125MB saved per forward pass at seq_len=2048
- [ ] No impact on output quality (logits identical for last position)
- [ ] Integrated into both standard and layer-by-layer inference paths

## Related Issues

- #1946 (layer-by-layer inference — applies to the lm_head step)
- #1942 (memory cleanup — 125MB savings reduces cleanup pressure)
